### PR TITLE
[adc_ctrl] documentation update

### LINKED ISSUE #10884: [adc_ctrl] Register adc_wakeup_status does not exist
Register adc_wakeup_status is mentioned a number of times in the design specification.
This register does not appear to exist in the RTL or adc_ctrl.hjson and thus UVM register model.

It seems that in the current RTL filter_status is masked by adc_wakeup_ctl to generate the wakeup signal.

### PR DESCRIPTION
**Pull request by tjaychen**
- clarify language around debounce counter
- clarify language around wakeup status
- companion PR to #10946 
- should fix #10884

Signed-off-by: Timothy Chen <timothytim@google.com>

### CODE REVIEW

**Comment by tjaychen** [comment]
@NigelScales let me know if this is clearer.

File: hw/ip/adc_ctrl/doc/_index.md [hunk 1]
```diff
@@ -81,22 +81,23 @@ Note that the time taken in this step depends on the properties of the ADC.
 
 In active operation the controller is in continuous scanning mode:
 * The ADC is continually powered on.
-* The sampling cycle time is the time taken for the ADC to take two samples (450us) plus NNN (fill in based on implementation but probably 4-10)) number of slow clock cycles x (5us; always-on clock frequency is 200KHz).
-* The debounce timer will trigger the filter status (and interrupt) after {{< regref "adc_sample_ctl" >}} multiplied by the continuous sampling cycle time.
+* The sampling cycle time is the time taken for the ADC to take two samples (450us) plus processing time from the adc controller..
+* The debounce timer will trigger the {{< regref "filter_status" >}} and interrupt after a configurable number of matching ADC samples, as determined by {{< regref "adc_sample_ctl" >}} have been seen.
```
Comment (matutem) [inline hw/ip/adc_ctrl/doc/_index.md:85]:
```suggestion
* The debounce timer will trigger the {{< regref "filter_status" >}} and interrupt after a configurable number of matching ADC samples have been seen, as determined by {{< regref "adc_sample_ctl" >}}.
```

File: hw/ip/adc_ctrl/doc/_index.md [hunk 2]
```diff
@@ -81,22 +81,23 @@ Note that the time taken in this step depends on the properties of the ADC.
 
 In active operation the controller is in continuous scanning mode:
 * The ADC is continually powered on.
-* The sampling cycle time is the time taken for the ADC to take two samples (450us) plus NNN (fill in based on implementation but probably 4-10)) number of slow clock cycles x (5us; always-on clock frequency is 200KHz).
-* The debounce timer will trigger the filter status (and interrupt) after {{< regref "adc_sample_ctl" >}} multiplied by the continuous sampling cycle time.
+* The sampling cycle time is the time taken for the ADC to take two samples (450us) plus processing time from the adc controller..
```
Comment (matutem) [inline hw/ip/adc_ctrl/doc/_index.md:84]: Line 72 above is missing a closing parenthesis.

**Review by matutem — APPROVED** [review]
LGTM, with a couple suggestions.

**Comment by NigelScales** [comment]
"Note that the debounce counter will remain at its limit until the set of filters are changed by software to debounce a different event."

I think this is incorrect.
It's the data which may change not the configuration.

File: hw/ip/adc_ctrl/doc/_index.md (hunk 2 above)
Comment (tjaychen) [reply hw/ip/adc_ctrl/doc/_index.md:84]: wow how did you see that?

File: hw/ip/adc_ctrl/doc/_index.md (hunk 1 above)
Comment (tjaychen) [reply hw/ip/adc_ctrl/doc/_index.md]: updated

File: hw/ip/adc_ctrl/doc/_index.md [hunk 3]
```diff
@@ -81,22 +81,23 @@ Note that the time taken in this step depends on the properties of the ADC.
...
 Once a small number of cycles have hit the filter with periodic scanning then the controller switches to continuous scanning in order to more accurately debounce the signal.
 In low power mode:
-* The ADC is only powered up for {{< regref "adc_pd_ctl.pwrup_time" >}} slow clock cycles plus the time taken for the ADC to take two samples plus NNN slow clock cycles.
-* The total time between samples for a periodic scan is the time the ADC is powered up plus {{< regref "adc_pd_ctl.wakeup_time" >}} cycles of the slow clock.
-* The time taken to transition to the continuous scanning mode is the total time of a slow scan multiplied by the {{< regref "adc_lp_sample_ctl" >}}.
-* The debounce timer will trigger the filter status (and interrupt) after the time taken to transition to fast mode plus {{< regref "adc_sample_ctl" >}} multiplied by the fast sampling cycle time.
+* The ADC is periodically powered up to take samples; this interval is determiend by {{< regref "adc_pd_ctl.wakeup_time" >}}.
+* Similar to normal operaiton, the ADC power-up delay is controlled by {{< regref "adc_pd_ctl.pwrup_time" >}}.
```
Comment (msfschaffner) [inline hw/ip/adc_ctrl/doc/_index.md:92]: nits: `operaiton -> operation`, `determiend -> determined` (on the line above)

File: hw/ip/adc_ctrl/doc/_index.md [hunk 4]
```diff
@@ -81,22 +81,23 @@ Note that the time taken in this step depends on the properties of the ADC.
...
+* Once the ADC is powered up, two samples are taken and compared to the filter thresholds.
+* If a configureable number of matches, as determined by {{< regref "adc_lp_sample_ctl" >}}, are seen, the adc controller transitions to normal operation for continuous sampling.
```
Comment (msfschaffner) [inline hw/ip/adc_ctrl/doc/_index.md:94]: nit: `configureable -> configurable`

**Review by msfschaffner — APPROVED** [review]

File: hw/ip/adc_ctrl/doc/_index.md (hunk 3 above)
Comment (tjaychen) [reply hw/ip/adc_ctrl/doc/_index.md]: thanks for noticing. I'll try to remember turning on spell check in the future...

File: hw/ip/adc_ctrl/doc/_index.md (hunk 4 above)
Comment (tjaychen) [reply hw/ip/adc_ctrl/doc/_index.md]: thanks for noticing. I'll try to remember turning on spell check in the future...

File: hw/ip/adc_ctrl/doc/_index.md (hunk 2 above)
Comment (matutem) [reply hw/ip/adc_ctrl/doc/_index.md]: I think from looking at the rendered document...

**Pull request merged**